emiproc/grids.py

```diff
@@ -761,3 +761,222 @@ def intersected_cells(self, corners):
+        """
+        self.dataset_path = dataset_path
+
+        with Dataset(dataset_path) as dataset:
+            self.clon_var = np.array(dataset["clon"][:]) * 180 / np.pi
+            self.clat_var = np.array(dataset["clat"][:]) * 180 / np.pi
+            self.cell_areas = np.array(dataset["cell_area"][:])
+            self.vlat = np.array(dataset["vlat"][:]) * 180 / np.pi
+            self.vlon = np.array(dataset["vlon"][:]) * 180 / np.pi
+            self.vertex_of_cell = np.array(dataset["vertex_of_cell"][:])
+            self.cell_of_vertex = np.array(dataset["cells_of_vertex"][:])
+
+
+        self.ncell = len(self.clat_var)
+
+        # Initiate a list of polygons, which is updated whenever the polygon of a cell
+        # is called for the first time
+        self.polygons = [None] * self.ncell
+
+        # Consider the ICON-grid as a 1-dimensional grid where ny=1
+        self.nx = self.ncell
+        self.ny = 1
+
+        super().__init__(name, ccrs.PlateCarree())
+
+
+    def cell_corners(self, n, j):
+        """Return the corners of the cell with index n.
+
+        Parameters
+        ----------
+        n : int
+        j : int
+
+        Returns
+        -------
+        tuple(np.array(shape=(3,), dtype=float),
+              np.array(shape=(3,), dtype=float))
+            Arrays containing the lon and lat coordinates of the corners
+        """
+
+        return self.vlon[self.vertex_of_cell[:,n]-1], self.vlat[self.vertex_of_cell[:,n]-1]
+
+
+    def indices_of_point(self, lon, lat, proj=ccrs.PlateCarree()):
+        """Return the indices of the grid cell that contains the point (lon, lat)
+
+        Parameters
+        ----------
+        lat : float
+            The latitude of the point source
+        lon : float
+            The longitude of the point source
+
+        Returns
+        -------
+        int
+            (icon_indn),
+            the index of the ICON grid cell containing the source.
+
+        Raises
+        ------
+        IndexError
+            If the point lies outside the grid.
+        """
+
+        indn = -1
+
+        pnt = Point(lon,lat)
+
+#       for n in np.arange(self.ncell):
+
+            
+        closest_vertex = ((self.vlon - lon)**2 + (self.vlat - lat)**2).argmin()
+        cell_range = self.cell_of_vertex[:,closest_vertex] - 1
+
+        for n in cell_range:
+
+            if n == -1:
+                continue
+
+#            icon_cell_xmax = max(self.vlon[self.vertex_of_cell[:,n]-1])
+#            if icon_cell_xmax < lon:
+#                continue
+#            icon_cell_xmin = min(self.vlon[self.vertex_of_cell[:,n]-1])
+#            if icon_cell_xmin > lon:
+#                continue
+#            icon_cell_ymax = max(self.vlat[self.vertex_of_cell[:,n]-1])
+#            if icon_cell_ymax < lat:
+#                continue
+#            icon_cell_ymin = min(self.vlat[self.vertex_of_cell[:,n]-1])
+#            if icon_cell_ymin > lat:
+#                continue
```

Comment:
Probably not necessary to keep these commented lines?

Reply:
Done

Reply:
Thanks